### ISSUE TITLE
How to use buffering with TlsStream?

### ISSUE THREAD

**Issue by david-monroe** [tokio-rs/tls#136]
I have seen that internally a large read buffer is used. Therefore I do not believe that it makes sense to ever use additional read buffering with TlsStream.

However, poll_write calls the underlying poll_write every time. Would it therefore make sense to use `TlsStream<BufWriter<TcpStream>>` or is this `BufWriter` unnecessary?

**Comment by quininer** [comment]
> poll_write calls the underlying poll_write every time.

I don't think this is correct, it only calls `poll_write` when it satisfies `wants_write`.

**Comment by david-monroe** [comment]
My interpretation was that `wants_write` returns true always if there is something to write at all.

**Comment by quininer** [comment]
You seem to be right. given that rustls has a flush method, this seems to be a reserved behavior for rustls.

Does rustls have any plan to expose write buffer? @ctz @djc

**Comment by djc** [comment]
We have a plan to have an unbuffered/write-through API, where rustls doesn't maintain any buffers for the data path.
